management/server/http/posture_checks_handler.go

```diff
@@ -199,27 +208,44 @@ func validatePostureChecksUpdate(req api.PostureCheckUpdate) error {
 		return status.Errorf(status.InvalidArgument, "posture checks name shouldn't be empty")
 	}
 
-	if req.Checks == nil || req.Checks.NbVersionCheck == nil {
+	if req.Checks == nil || (req.Checks.NbVersionCheck == nil && req.Checks.OsVersionCheck == nil) {
 		return status.Errorf(status.InvalidArgument, "posture checks shouldn't be empty")
 	}
 
 	if req.Checks.NbVersionCheck != nil && req.Checks.NbVersionCheck.MinVersion == "" {
 		return status.Errorf(status.InvalidArgument, "minimum version for NetBird's version check shouldn't be empty")
 	}
 
+	if osVersionCheck := req.Checks.OsVersionCheck; osVersionCheck != nil &&
+		osVersionCheck.Android != nil && osVersionCheck.Android.MinVersion == "" &&
+		osVersionCheck.Darwin != nil && osVersionCheck.Darwin.MinVersion == "" &&
+		osVersionCheck.Ios != nil && osVersionCheck.Ios.MinVersion == "" &&
+		osVersionCheck.Linux != nil && osVersionCheck.Linux.MinKernelVersion == "" &&
+		osVersionCheck.Windows != nil && osVersionCheck.Windows.MinKernelVersion == "" {
+		return status.Errorf(status.InvalidArgument, "minimum version for at least one OS in the OS version check shouldn't be empty")
+	}
```

Comment:
Consider triggering an error if a single OS version check is set, but its corresponding requirement is not set. Currently, an error is only triggered when all OS versions are set and empty.

```suggestion
	if osVersionCheck := req.Checks.OsVersionCheck; osVersionCheck != nil &&
		osVersionCheck.Android != nil && osVersionCheck.Android.MinVersion == "" ||
		osVersionCheck.Darwin != nil && osVersionCheck.Darwin.MinVersion == "" ||
		osVersionCheck.Ios != nil && osVersionCheck.Ios.MinVersion == "" ||
		osVersionCheck.Linux != nil && osVersionCheck.Linux.MinKernelVersion == "" ||
		osVersionCheck.Windows != nil && osVersionCheck.Windows.MinKernelVersion == "" {
		return status.Errorf(status.InvalidArgument, "minimum version for at least one OS in the OS version check shouldn't be empty")
	}
```

Reply:
Good caught! I've changed the conditions but split the conditions for easy readability. Also, added tests to cover the cases for this function.